glean-core/src/metrics/experiment.rs

```diff
@@ -18,9 +18,25 @@ use crate::Lifetime;
 // An internal ping name, not to be touched by anything else
 const INTERNAL_STORAGE: &str = "glean_internal_info";
 
+// The following values are maximums related to experiments, these were
+// chosen based on current desktop usage. See here, for reference:
+// https://searchfox.org/mozilla-central/rev/3366c3d24f1c3818df37ec0818833bf085e41a53/toolkit/components/telemetry/app/TelemetryEnvironment.jsm#58
+
 /// The maximum length of the experiment id and the branch id. Identifiers
 /// longer than this number of characters are truncated.
-const MAX_EXPERIMENTS_IDS_LEN: usize = 30;
+const MAX_EXPERIMENTS_IDS_LEN: usize = 100;
+/// The maximium length of the experiment branch id.  Any branch id longer
+/// than this will be truncated.
+const MAX_EXPERIMENTS_BRANCH_LEN: usize = 100;
+/// The maximum number of extras allowed in the extras hash map.  Any items
+/// added beyond this limit will be dropped.
+const MAX_EXPERIMENTS_EXTRAS_LEN: usize = 20;
+/// The maximum length of the experiment extra keys.  Keys and values longer
+/// than this limit will be truncated.
+const MAX_EXPERIMENT_KEY_LEN: usize = 50;
```

Comment:
Let's use the same `MAX_EXPERIMENTS_IDS_LEN` here, for consistency? On desktop, I don't see things that need 50 (I only see `type` and `enrollmentID` being used), so 30 should be enough.